db/sql/deploy/buildings/functions/buildings_update_suburb_locality.sql

```diff
@@ -0,0 +1,141 @@
+-- Deploy nz-buildings:buildings/functions/update_suburb_locality_changed_deleted_buildings to pg
+
+BEGIN;
+
+-- building_outlines_update_changed_and_deleted_suburb (replace suburb values with the intersection result)
+    -- params: 
+    -- return: integer count of number of building outlines updated
+
+CREATE OR REPLACE FUNCTION buildings.building_outlines_update_changed_and_deleted_suburb()
+RETURNS integer AS
+$$
+    -- Find new ids and geometry differences of changed 
+    WITH changed_geometries
+    AS (SELECT
+        nzl.id AS new_external_id,
+        bsl.suburb_locality_id AS new_id,
+        ST_Difference(ST_Transform(nzl.shape, 2193),bsl.shape) AS shape
+    FROM admin_bdys.nz_locality nzl,
+        buildings_reference.suburb_locality bsl
+    WHERE 
+        bsl.external_suburb_locality_id = nzl.id
+        AND (NOT ST_Equals(ST_Transform(nzl.shape, 2193), bsl.shape))
+        AND (NOT st_isempty(ST_Difference(ST_Transform(nzl.shape, 2193),bsl.shape)))),
+    
+    -- Joins the difference and new ids with the old suburb_locality_id
+    intersected_change
+    AS (SELECT 
+        bsl.external_suburb_locality_id AS old_id,
+        cg.new_external_id,
+        cg.new_id,
+        cg.shape AS shape
+    FROM buildings_reference.suburb_locality bsl,
+         changed_geometries cg
+    WHERE 
+        (ST_Overlaps(cg.shape, bsl.shape) OR ST_Within(cg.shape, bsl.shape))
+        AND bsl.external_suburb_locality_id != cg.new_external_id),
+    
+    -- Updates building outlines where:
+      -- It's external id is an 'old id'
+      -- It Overlaps or is Within the difference polygon for the 'old id'
+      -- It overlaps the difference polygon by the largest proportion
+    building_outline_updates
+    AS (UPDATE 
+        buildings.building_outlines
+        SET suburb_locality_id = ic.new_id
+    FROM 
+    intersected_change ic, buildings_reference.suburb_locality bsl
+    WHERE 
+        buildings.building_outlines.suburb_locality_id = bsl.suburb_locality_id
+        AND bsl.external_suburb_locality_id = ic.old_id
+        AND (ST_Within(ST_Expand(buildings.building_outlines.shape, 100), ic.shape)
```

Comment:
We previously used ST_Expand 1000m.  Consider changing back to this or ensure all buildings are within 100m of LOCALITY/ISLAND/SUBURB/PARK_RESERVE